xamarin/Xamarin.Forms
# Remove experimental flag for shell and change a few apis

### Description of Change ###

Remove the experimental flag, rename all the apis that are Shell.GetShell, and internalize some types

### Issues Resolved ### 
- fixes #2415

### API Changes ###
Changed:
 - Shell.ShellBackgroundColor => Shell.BackgroundColor
 - Shell.ShellTitleColor=> Shell.TitleColor
 - Shell.ShellDisabledColor => Shell.DisabledColor
 - Shell.ShellUnselectedColor => Shell.UnselectedColor
 - Shell.ShellForegroundColor => Shell.ForegroundColor

 Removed:
 - ShellItemCollection (made internal)
- ShellSectionCollection (made internal)
- ShellContentCollection (made internal)
- ShellSection.NavigationImpl (made internal)
- Shell.NavigationImpl (made internal)

 
### Platforms Affected ### 
- Core/XAML (all platforms)


### Testing Procedure ###
- make sure shell works
- look through shell, shell section, shell item, shell content and see if there's anything else we should hide/expose

### PR Checklist ###
